### PR TITLE
MAINT: Deprecate link aliases

### PR DESCRIPTION
**Pull request by bashtage**
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by josef-pkt** [comment]
Yes, I'm all in favor, #8323

**Comment by josef-pkt** [comment]
another cleanup would be to rename all lower case (sub)classes that are not duplicates/aliases to standard python class names with capitalized first letter.
But that at least does not mess up the documentation.

File: statsmodels/genmod/tests/test_glm.py [hunk 1]
```diff
@@ -301,9 +301,7 @@ def test_score_test(self):
         assert_equal(df, 1)
 
     def test_get_prediction(self):
-        with pytest.warns(DeprecationWarning):
-            # deprecation warning for linear keyword
-            pred1 = self.res1.get_prediction()  # GLM
```
Comment (josef-pkt) [inline statsmodels/genmod/tests/test_glm.py:306]: why did you remove these?
I just added them a while ago.

File: statsmodels/base/_prediction_inference.py [hunk 2]
```diff
@@ -456,7 +456,7 @@ def get_prediction_glm(self, exog=None, transform=True,
     if pred_kwds is None:
         pred_kwds = {}
 
-    pred_kwds['linear'] = False
+    # pred_kwds['linear'] = False
```
[the remark this answers is not in the document]
Comment (bashtage) [reply statsmodels/base/_prediction_inference.py]: Isn't the `linear` is deprecated warning old?

File: statsmodels/discrete/tests/test_discrete.py [hunk 3]
```diff
@@ -1646,8 +1646,7 @@ def test_negative_binomial_default_alpha_param():
     with warnings.catch_warnings():
         warnings.simplefilter("error")
         sm.families.NegativeBinomial(alpha=1.0)
-    with warnings.catch_warnings():
-        warnings.simplefilter("error")
```
[the remark this answers is not in the document]
Comment (josef-pkt) [reply statsmodels/discrete/tests/test_discrete.py:1654]: The warning for negbin is a UserWarning, AFAIR
so this could be more selective to handle both kinds of warning

I guess the following will only raise the UserWarning

sm.families.NegativeBinomial(link=sm.families.links.NegativeBinomial(alpha=1.0),
                                     alpha=1.0)

**Comment by josef-pkt** [comment]
In the family module, the families have link attributes which should use the new names
e.g. for Binomial family
`links = [L.Logit, L.probit, L.cauchy, L.Log, L.LogC, L.CLogLog, L.LogLog,
             L.identity]`

I'm waiting for this to be merged, then I will make a PR to make link check/exception optional. That would have many merge conflicts with the correction of families `links` attributes

**Comment by bashtage** [comment]
Same buggy test failed here as well.

**Comment by josef-pkt** [comment]
I will cleanup some warnings.

I had 2 warnings for all genmod unit tests a few days ago, now there are 149 warnings

**Comment by josef-pkt** [comment]
@bashtage 
Why did you use FutureWarning instead of DeprecationWarning?
any specific reason?
